packages/amplify-category-storage/src/index.ts

```diff
@@ -1,76 +1,103 @@
     return;
   }
 
-  return providerController.addResource(context, categoryName, service, options);
+  return providerController.addResource(context, AmplifyCategories.STORAGE, service, options);
 }
 
-export async function console(context: $TSContext) {
-  printer.info(`to be implemented: ${categoryName} console`);
+async function categoryConsole(context: any) {
+  printer.info(`to be implemented: ${AmplifyCategories.STORAGE} console`);
 }
 
-export async function migrate(context: $TSContext) {
+async function migrateStorageCategory(context: any) {
   const { projectPath, amplifyMeta } = context.migrationInfo;
-  const migrateResourcePromises: Promise<$TSAny>[] = [];
-
-  const categoryResources = amplifyMeta?.[categoryName] || {};
+  const migrateResourcePromises: any = [];
+
+  Object.keys(amplifyMeta).forEach(categoryName => {
+    if (categoryName === AmplifyCategories.STORAGE) {
+      Object.keys(amplifyMeta[AmplifyCategories.STORAGE]).forEach(resourceName => {
+        try {
+          const providerController = require(`./provider-utils/${amplifyMeta[AmplifyCategories.STORAGE][resourceName].providerPlugin}`);
+
+          if (providerController) {
+            migrateResourcePromises.push(
+              providerController.migrateResource(
+                context,
+                projectPath,
+                amplifyMeta[AmplifyCategories.STORAGE][resourceName].service,
+                resourceName,
+              ),
+            );
+          } else {
+            printer.error(`Provider not configured for ${AmplifyCategories.STORAGE}: ${resourceName}`);
+          }
+        } catch (e) {
+          printer.warn(`Could not run migration for ${AmplifyCategories.STORAGE}: ${resourceName}`);
+          throw e;
+        }
+      });
+    }
+  });
 
-  for (const resourceName of Object.keys(categoryResources)) {
-    try {
-      const providerController = await import(`./provider-utils/${amplifyMeta[categoryName][resourceName].providerPlugin}`);
+  await Promise.all(migrateResourcePromises);
+}
 
-      if (providerController) {
-        migrateResourcePromises.push(
-          await providerController.migrateResource(context, projectPath, amplifyMeta[categoryName][resourceName].service, resourceName),
-        );
-      } else {
-        printer.error(`Provider not configured for ${categoryName}: ${resourceName}`);
-      }
-    } catch (e) {
-      printer.warn(`Could not run migration for ${categoryName}: ${resourceName}`);
-      throw e;
+async function transformCategoryStack(context: $TSContext, resource: IAmplifyResource) {
+  if (resource.service === 'DynamoDB') {
+    if (canResourceBeTransformed(resource.resourceName)) {
+      const stackGenerator = new DDBStackTransform(resource.resourceName);
+      stackGenerator.transform();
     }
+  } else if (resource.service === 'S3') {
+    // Not yet implemented
   }
+}
 
-  await Promise.all(migrateResourcePromises);
+function canResourceBeTransformed(resourceName: string) {
```

Comment:
since this is a DDB function, please name it as such.

Reply:
This can be re-used for the S3 service as well if you'd like when you integrate S3's migration logic in here - hence named it generic.